src/streaming-ingestion-stack.ts

```diff
@@ -0,0 +1,168 @@
+  public readonly redshiftProvisionedStack?: StreamingIngestionRedshiftStack;
+
+  constructor(
+    scope: Construct,
+    id: string,
+    props?: StackProps,
+  ) {
+    super(scope, id, props);
+
+    const featureName = 'StreamingIngestion';
+    this.templateOptions.description = `(${SolutionInfo.SOLUTION_ID}-si) ${SolutionInfo.SOLUTION_NAME} - ${featureName} ${SolutionInfo.SOLUTION_VERSION_DETAIL}`;
+
+    const p = createStackParameters(this);
+    this.templateOptions.metadata = p.metadata;
+
+    const streamModeStr = p.params.kinesis.mode;
+    const redshiftModeStr = p.params.redshift.mode;
+
+    const kinesisOnDemandRedshiftServerlessStackCondition = new CfnCondition(
+      this,
+      'kinesisOnDemandRedshiftServerlessStackCondition',
+      {
+        expression:
+          Fn.conditionAnd(
+            Fn.conditionEquals(streamModeStr, 'ON_DEMAND'),
+            Fn.conditionEquals(redshiftModeStr, REDSHIFT_MODE.SERVERLESS),
+          ),
+      },
+    );
+
+    const kinesisProvisionedRedshiftServerlessStackCondition = new CfnCondition(
+      this,
+      'kinesisProvisionedRedshiftServerlessStackCondition',
+      {
+        expression:
+          Fn.conditionAnd(
+            Fn.conditionEquals(streamModeStr, 'PROVISIONED'),
+            Fn.conditionEquals(redshiftModeStr, REDSHIFT_MODE.SERVERLESS),
+          ),
+      },
+    );
+
+    const kinesisOnDemandRedshiftProvisionedStackCondition = new CfnCondition(
+      this,
+      'kinesisOnDemandRedshiftProvisionedStackCondition',
+      {
+        expression:
+          Fn.conditionAnd(
+            Fn.conditionEquals(streamModeStr, 'ON_DEMAND'),
+            Fn.conditionEquals(redshiftModeStr, REDSHIFT_MODE.PROVISIONED),
+          ),
+      },
+    );
+
+    const kinesisProvisionedRedshiftProvisionedStackCondition = new CfnCondition(
+      this,
+      'kinesisProvisionedRedshiftProvisionedStackCondition',
+      {
+        expression:
+          Fn.conditionAnd(
+            Fn.conditionEquals(streamModeStr, 'PROVISIONED'),
+            Fn.conditionEquals(redshiftModeStr, REDSHIFT_MODE.PROVISIONED),
+          ),
+      },
+    );
+
+    const stackProps = {
+      projectId: p.params.projectId,
+      appIds: p.params.appIds,
+      kinesis: {
+        kinesisStreamMode: p.params.kinesis.mode,
+        kinesisDataRetentionHours: p.params.kinesis.dataRetentionHours,
+        kinesisShardCount: p.params.kinesis.shardCount,
+      },
+      pipeline: {
+        kinesisSourceStreamName: p.params.pipeline.kinesisSourceStreamName,
+        parallelism: p.params.pipeline.parallelism,
+        parallelismPerKPU: p.params.pipeline.parallelismPerKPU,
+        applicationCodeBucketARN: p.params.pipeline.applicationCodeBucketARN,
+        applicationCodeBucketPrefix: p.params.pipeline.applicationCodeBucketPrefix,
+        securityGroupIds: p.params.pipeline.securityGroupIds,
+        subnetIds: p.params.pipeline.subnetIds,
+      },
```

Comment:
can we simplify it with `pipeline: p.params.pipeline`?

Reply:
done